libs/indibase/indiweatherinterface.cpp

```diff
@@ -187,6 +187,43 @@ bool WeatherInterface::setCriticalParameter(std::string param)
     return false;
 }
 
+IPState WeatherInterface::checkParameterState(INumber parameter)
```

Comment:
IMO, this should be

IPState checkParameterState(const INumber *parameter) const

Reply:
Hi, thanks for the review.

Making both methods const is a nice addon as a safety measure against unintended changes in internal state.

But I'm not sure I understand why passing the INumber as a pointer and not a reference like the string version is desireable.

Reply:
You can also pass by reference.

Reply:
Thanks.
I updated this pull request with those changes.